tests/sycl/test_sycl_allocators.cpp

```diff
@@ -0,0 +1,78 @@
+/** VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "vecmem/allocators/allocator.hpp"
+#include "vecmem/memory/memory_manager.hpp"
+#include "vecmem/memory/sycl/direct_memory_manager.hpp"
+
+// System include(s).
+#undef NDEBUG
```

Comment:
This seems odd. I guess it's not dangerous in a test, but what's the reason for this?

Reply:
Got it, so that `assert` works even with a release build.

Reply:
We do this in quite a few places in the Athena core software...

I didn't want to start with something like gtest right away. But that's absolutely in the cards.

Reply:
Cool.

Reply:
I personally define a few ASSERT macros for this situation, but fair enough.

Reply:
The reason that I came to like using the system `assert(...)` function is that it very nicely works in CUDA device code as well. (All in all using `assert(...)` in CUDA code proved to be a godsend while I was working on the seedfinder code.)

Reply:
That's a good point. I guess the structure and reporting of something like GTest can still provide a net benefit, at least on the host.